Final/LinkedList.cpp

```diff
@@ -0,0 +1,82 @@
+#include "LinkedList.h"
+
+#include <iostream>
+#include <stdlib.h>
+
+// Create linked list
+LinkedList::LinkedList()
+{
+    head = sentinel = (node *)calloc(1, sizeof(node));
+}
+
+// Clean up nodes
+LinkedList::~LinkedList()
+{
+    node *cur = head;
+    while (cur)
+    {
+        node *temp = cur->next;
+        free(cur);
+        cur = temp;
+    }
+}
+
+// Adds node to list
+void LinkedList::push(int data)
+{
+    node *new_node = (node *) malloc(sizeof(node));
+    new_node->data = data;
+    new_node->next = head;
+    head = new_node;
+}
+
+// Removes node from list
+void LinkedList::remove(int data)
+{
+    node *pre;
+    node *cur = head;
+    while (cur->data != data)
+    {
+        pre = cur;
+        cur = cur->next;
+        if(!cur) // node does not exist
+            return;
+    }
+    
+    if (cur == head)
+        head = cur->next;
+    else if (cur == sentinel)
```

Comment:
the sentinel is a dummy node used for iteration. you should never remove the sentinel

Reply:
https://en.wikipedia.org/wiki/Sentinel_node

Reply:
Ohhh that’s what I was so confused about, cause I thought it was pointing to the last node and I was confused how it was getting set and all that.